osu.Game/Screens/OnlinePlay/Match/RoomModSelectOverlay.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Game.Online.Rooms;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Mods;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Mods;
+
+namespace osu.Game.Screens.OnlinePlay.Match
+{
+    public partial class RoomModSelectOverlay : UserModSelectOverlay
+    {
+        [Resolved]
+        private IBindable<PlaylistItem> selectedItem { get; set; } = null!;
+
+        [Resolved]
+        private RulesetStore rulesets { get; set; } = null!;
+
+        private readonly List<Mod> roomMods = new List<Mod>();
+
+        public RoomModSelectOverlay()
+            : base(OverlayColourScheme.Plum)
+        {
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            selectedItem.BindValueChanged(_ =>
+            {
+                roomMods.Clear();
+
+                if (selectedItem.Value is PlaylistItem item)
+                {
+                    var rulesetInstance = rulesets.GetRuleset(item.RulesetID)?.CreateInstance();
+                    Debug.Assert(rulesetInstance != null);
+                    roomMods.AddRange(item.RequiredMods.Select(m => m.ToMod(rulesetInstance)));
+                }
+
+                SelectedMods.TriggerChange();
```

Comment:
I also don't like the way this proxy-launches flows via the `TriggerChange()` methods, I'd just plainly call `UpdateOverlayInformation()` here rather than do this. Or at least add an inline comment what this is trying to do.

Reply:
Right, that was my intention but I missed doing that.